libraries/resource_provider.rb

```
@@ -183,6 +183,46 @@ def create(method = :create)
       end
     end
 
+    # Adds Oneview Scope to the Oneview resource if scope is not already added
+    def add_scope
```

My thoughts here:
  1.  You should support both adding to single or multiple scopes too (same for removing)
  2. It would be better to call the method and actions 'add_to_scope'

---

ok, but this generate 2 things:
1 - each scope of scopes list will be added one per time, if some problem occurs the operation could be not completed and some scopes could be added and others not. In this case, what do you think which should we do? Should we maintain resource consistence after the that problems? Or should we just to throw the error?

---

For add, you should keep the existing scopes, and add just the new ones.
For remove, you should keep the existing scopes, and add just the new ones.

This way you will keep the action idempotent.